scylla-server-rust/src/reciever/mqtt_reciever.rs

```diff
@@ -0,0 +1,175 @@
+                // parse the message into the data and the node name it falls under
+                let item_data = match self.parse_msg(msg).await {
+                    Ok(msg) => msg,
+                    Err(err) => {
+                        warn!("Message parse error: {:?}", err);
+                        continue;
+                    }
+                };
+                // send the message over the channel to the socketio and database consumers
+                self.send_msg(item_data).await;
+            }
+        }
+    }
+
+    /// Parse the message
+    /// * `msg` - The mqtt message to parse
+    /// returns the ClientData, or the Err of something that can be debug printed
+    #[instrument(skip(self), level = Level::TRACE)]
+    async fn parse_msg(&self, msg: Publish) -> Result<ClientData, impl fmt::Debug> {
+        let data = serverdata::ServerData::parse_from_bytes(&msg.payload)
+            .map_err(|f| format!("Could not parse message topic:{:?} error: {}", msg.topic, f))?;
+
+        let split = std::str::from_utf8(&msg.topic)
+            .unwrap_or_else(|_| panic!("Could not parse topic: {:?}", msg.topic))
+            .split_once('/')
+            .ok_or(&format!("Could not parse nesting: {:?}", msg.topic))?;
+
+        let node = split.0;
+
+        let data_type = split.1.replace('/', "-");
+
+        // extract the unix time, returning the current time instead if needed
+        // note the Cow magic involves the return from the map is a borrow, but the unwrap cannot as we dont own it
+        let unix_time = msg
+            .properties
+            .unwrap_or_default()
+            .user_properties
+            .iter()
+            .map(Cow::Borrowed)
+            .find(|f| f.0 == "ts")
+            .unwrap_or_else(|| {
+                debug!("Could not find timestamp in mqtt, using current time");
+                Cow::Owned((
+                    "ts".to_string(),
+                    chrono::offset::Utc::now().timestamp_millis().to_string(),
+                ))
+            })
+            .into_owned();
+
+        // parse time, if time isnt present use sys time (see above)
+        let Ok(time_clean) = unix_time.1.parse::<i64>() else {
+            return Err(format!("Invalid timestamp: {}", unix_time.1));
+        };
+        // ts check for bad sources of time which may return 1970
+        if time_clean < 963014966000 {
+            return Err(format!("Timestamp before year 2000: {}", unix_time.1));
+        }
+
+        Ok(ClientData {
+            run_id: self.curr_run,
+            name: data_type,
+            unit: data.unit,
+            values: data.value,
+            timestamp: time_clean,
+            node: node.to_string(),
+        })
+    }
+
+    /// Send a message to the channel, printing and IGNORING any error that may occur
+    /// * `client_data` - The cliet data to send over the broadcast
+    async fn send_msg(&self, client_data: ClientData) {
```

Comment:
feels a little weird calling this just a receiver if it's in charge of receiving mqtt messages and sending data to client (if I'm reading this right)

Reply:
I refactored to "processor" lmk if that sounds better.

Reply:
Maybe manager? Like mqtt_manager, db_manager, mock_mqtt_manager (or mqtt could be socket_manager, mock_socket_manager up to you). Processor just feels like such a generic name, and manager describes what the service is doing (managing the connection with another process).

Reply:
Ultimately do whatever seems most intuitive to you, this is kind of a pedantic comment anyways